pkg/scheduler/core/generic_scheduler.go

```diff
@@ -269,10 +269,17 @@ func (g *genericScheduler) Preempt(ctx context.Context, prof *profile.Profile, s
 	}
 	potentialNodes := nodesWherePreemptionMightHelp(allNodes, fitError)
 	if len(potentialNodes) == 0 {
-		klog.V(3).Infof("Preemption will not help schedule pod %v/%v on any node.", pod.Namespace, pod.Name)
+		klog.V(5).Infof("Preemption will not help schedule pod %v/%v on any node.", pod.Namespace, pod.Name)
 		// In this case, we should clean-up any existing nominated node name of the pod.
 		return "", nil, []*v1.Pod{pod}, nil
 	}
+	if klog.V(5).Enabled() {
+		var sample []string
+		for i := 0; i < int(math.Min(10, float64(len(potentialNodes)))); i++ {
```

Comment:
I would just do `i < 10 && i < len(potentialNodes)` to avoid float conversions

Reply:
done. there should be max/min functions for different types in math.

Reply:
https://groups.google.com/g/golang-nuts/c/dbyqx_LGUxM :)